tools/gmhazard_scripts/db_creation/empirical_db/new_calc_emp_ds.py

```diff
@@ -101,95 +90,112 @@ def calculate_ds(
+                                nhm_data,
+                                classdef.TectType[tect_type],
+                            )
+                            gmm_calculated_df = openquake_wrapper_vectorized.oq_run(
+                                GMM,
+                                classdef.TectType[tect_type],
+                                rupture_context_df,
+                                str(im),
+                                psa_periods if im is gc.im.IMType.pSA else None,
+                            )
+                            # Matching the index with rupture_df
+                            # to have a right rupture label
+                            gmm_calculated_df.set_index(
+                                rupture_df[
+                                    rupture_df["rupture_name"].isin(
+                                        rupture_context_df["rupture_name"]
+                                    )
+                                ].index,
+                                inplace=True,
+                            )
+                            gmm_calculated_df.columns = [
+                                f"{'_'.join(col.split('_')[:2]) if im is gc.im.IMType.pSA else im}_sigma"
+                                if col.endswith("std_Total")
+                                else col
+                                for col in gmm_calculated_df
+                            ]
+
+                            # Write an im_df to the given station/site
+                            imdb.add_im_data(
+                                site,
+                                gmm_calculated_df.loc[
+                                    :,
+                                    gmm_calculated_df.columns.str.endswith(
+                                        ("mean", "sigma")
+                                    ),
+                                ],
+                            )
+
+                        print(f"Writing metadata for Model: {GMM.name}")
+                        imdb.write_sites(site_df[["lat", "lon"]])
+                        imdb.write_rupture_data(rupture_df)
+                        im_list = common.curate_im_list(
+                            yaml.safe_load(open(model_dict_ffp)), db_type, psa_periods
                         )
-                        imdb_dict[db_type].close()
-
-    print("Writing metadata")
-    common.write_metadata_and_close(
-        imdb_dict,
-        background_sources_ffp,
-        rupture_df,
-        site_df,
-        vs30_ffp,
-        psa_periods,
-        ims,
-        model_dict_ffp,
-    )
-    print("Done")
+                        imdb.write_attributes(
+                            Path(background_sources_ffp).name,
+                            Path(vs30_ffp).name,
+                            ims=np.asarray(im_list, dtype=str),
+                        )
+                        print(f"Writing metadata for Model: {GMM.name} is done.")
```

Comment:
I'd argue to still use the function (just for consistency sake) - probably best to either create a sub_function that writes metadata for one db or to call the existing function with a dummy dictionary for the single db.

Reply:
Just a thought, shall we implement this in a different PR? :thinking:

Reply:
Should be fine in this PR as existing functionality shouldn't be affected.

Reply:
@jasonmotha Assuming I'm not missing something here, I recon the way Tom has done it is good, it's nice and tidy imo. The IMDB is open for the for the minimal amount of time vs keeping it open (or opening it again). And it doesn't requires having to keep track of the IMDBs in a dictionary (i.e process and forget).

Reply:
@SnowNooDLe @jasonmotha From my understanding `common.write_metadata_and_close()` is a different approach. It assumes that all IMDBs are opened initially and then kept open throughout the script and then after processing `common.write_metadata_and_close()` is called to write metadata to all of them and then close them all (which was needed for the previous MPI version). 
Whereas what you have written above, only opens each IMDB when needed, writes the data (and metadata) and then closes it before moving onto the next, which imo is cleaner. 
If we are also going do tidy up, then I recon we should bring `calc_emp_flt.py` in line with what you have above, which I think is the only other usage of `common.write_metadata_and_close()`